docs/guides/code-style-guide.md

```diff
@@ -0,0 +1,161 @@
+
+Some of the guidelines are so common that there are automatic tools to do it for you! One of these tools is called Prettier. You should have installed this tool when you [set up your VS Code extensions](https://docs.codeyourfuture.io/course-processes/before-the-course/getting-setup#extensions).
+
+Prettier is a tool that rearranges your code (called *formatting*). It follows a set of rules that programmers have agreed make your code easier to read and understand.
+
+You might have a common problem when running Prettier. If you see something like this:
+
+![Invalid code with red highlight](./assets/prettier-error.png)
+
+Then you have written some code that is invalid and Prettier can't format it. A common is a missing bracket, so try looking for that first.
+
+## Indent your code
+
+You may have noticed that programmers tend to *indent* their code by starting a line of code with some space.
+
+Here's an example in HTML:
+
+```html
+<div>
+  <p>Hello world!</p>
+</div>
+```
+
+And JavaScript:
+
+```js
+if (isVegetarian) {
+  return "Macaroni and Cheese";
+} else {
+  return "Steak and Chips";
+}
+```
+
+Indentation helps us to read our code by visually marking how the code is going to run. Programmers rely on this extra information to quickly read and understand code.
+
+There are some guidelines about where to add indentation. For example, lines of code are indented when they are inside a function, an `if` statement or a `for` loop. Additionally, if any of those structures are inside another structure, then they are indented further.
+
+The guidelines about where to add indentation can be complex, but the good news is that **Prettier adds indentation automatically** for us! However it is good to pay attention to where indentation is added, as you may find yourself writing code in an environment where Prettier is not installed.
+
+## Don't leave lots of commented out code
+
+When you are debugging a problem, you might comment out some sections of your code.
+
+```js
+// 🛑 Don't do this!
+function addToShoppingList(item) {
+  // console.log("Shopping list before", shoppingList);
+  // console.log("Adding item", item);
+  shoppingList.add(item);
+  // console.log("Shopping list after", shoppingList);
+}
+```
+
+In this example, there is some extra debugging information that is commented out.
+
+Once you have solved the problem, you should remove large sections of commented out code. When you (or someone else) are reading your code, you want to know only the important pieces. Removing commented out code helps find the relevant code faster and easier.
+
+### Saving old versions with Git
+
+You might be worried about "losing" some experimental code that you want to "save" by commenting it out. However, this can cause confusion about which bits of code still work.
+
+If you want to save some code, remember that you can commit it using Git. It will then be saved it forever. If decide to remove the code later you can delete it and make another commit.
```

Comment:
```suggestion
If you want to save some code, remember that you can commit it using Git. It will then be saved it forever. If you decide to remove the code later you can delete it and make another commit. And if you want it back again, you can always look at the deleted code in the git history.
```

Reply:
also small typo: 'It will then be saved it forever' in this block